src/pipx/commands/install.py

```diff
@@ -33,73 +33,86 @@ def install(
 
     python = python or DEFAULT_PYTHON
 
-    if package_name is None:
-        package_name = package_name_from_spec(package_spec, python, pip_args=pip_args, verbose=verbose)
-    if venv_dir is None:
-        venv_container = VenvContainer(constants.PIPX_LOCAL_VENVS)
-        venv_dir = venv_container.get_venv_dir(f"{package_name}{suffix}")
+    if package_names is None:
+        package_names = [
+            package_name_from_spec(package_spec, python, pip_args=pip_args, verbose=verbose)
+            for package_spec in package_specs
+        ]
+    elif len(package_names) != len(package_specs):
+        package_names = [
+            package_name_from_spec(package_spec, python, pip_args=pip_args, verbose=verbose)
+            for package_spec in package_specs
+        ]
 
-    try:
-        exists = venv_dir.exists() and bool(next(venv_dir.iterdir()))
-    except StopIteration:
-        exists = False
+    for package_name, package_spec in zip(package_names, package_specs, strict=True):
```

Comment:
Probably telling the linter not to bother about a missing `strict` argument until we have dropped these Python versions is enough here?

Don't know what error code this generates, but you could write a `# noqa: EXYZ`.

Reply:
Thanks for your help! Just removing the `strict=True` and re-running the linter without any cached nox environments etc. seems to now pass, without needing to use any `# noqa` for some reason?

Either way, linting and tests (including on 3.8 etc.) seem to now pass locally.